wgpu-types/src/lib.rs

```diff
@@ -3991,39 +3991,57 @@ impl<T> Maintain<T> {
 
     /// Map on the wait index type.
     #[must_use]
-    pub fn map_index<U, F>(self, func: F) -> Maintain<U>
+    pub fn map_index<U, F>(self, func: F) -> PollType<U>
     where
         F: FnOnce(T) -> U,
     {
         match self {
-            Self::WaitForSubmissionIndex(i) => Maintain::WaitForSubmissionIndex(func(i)),
-            Self::Wait => Maintain::Wait,
-            Self::Poll => Maintain::Poll,
+            Self::WaitForSubmissionIndex(i) => PollType::WaitForSubmissionIndex(func(i)),
+            Self::Wait => PollType::Wait,
```

Comment:
I'm kinda tempted to take the opportunity and bake the "non-webgpu-ness" of those wait operations into the names, but I don't know how to do so without making things worse 🤷

Reply:
Yeah... the behavior on webgpu has always been a bit cheeky...

Idk :)